front/components/vaults/VaultResourcesList.tsx

```diff
@@ -176,18 +195,96 @@ export const VaultResourcesList = ({
       onClick: () => onSelect(r.sId),
     })) || [];
 
-  if (isDataSourcesLoading || isVaultDataSourceViewsLoading) {
+  if (
+    isDataSourcesLoading ||
+    isVaultDataSourceViewsLoading ||
+    isSystemVaultDataSourceViewsLoading
+  ) {
     return (
       <div className="mt-8 flex justify-center">
         <Spinner size="lg" />
       </div>
     );
   }
 
-  const setUpDataSources = vaultDataSourceViews.map((dsv) => dsv.connectorId);
-  const unusedDataSources = managedDataSources.filter(
-    (ds) => !setUpDataSources.includes(ds.connectorId)
-  );
+  const saveSelectedDatasources = async (
```

Comment:
Same this should be in the component, remember this component is a simple list 😉